### PR TITLE
Port patches from 1.0.0 branch

### PR DESCRIPTION
**Pull request by hcho3**
* **Remove f-string, since it's not supported by Python 3.5 (#5330)**. This necessitated 1.0.1 patch release.
* Bump version to 1.1.0-SNAPSHOT, to prepare for next release

Also, when I was working on 1.0.1 patch release, I fixed two bugs:
* Enforce only major version in JSON model schema: The schema had 1.0.0 version hard-coded in, so any bump in the version in the codebase would cause schema violation. I revised the schema to only fix the major version. Any major revision of the model schema would bump up the major version.
* Fix a unit test that mistook MINOR ver for PATCH ver: This appears to be a typo in the unit test.

### CODE REVIEW

File: python-package/xgboost/sklearn.py [hunk 1]
```diff
@@ -423,8 +423,8 @@ def load_model(self, fname):
                 self.classes_ = np.array(v)
                 continue
             if k == 'type' and type(self).__name__ != v:
-                msg = f'Current model type: {type(self).__name__}, ' + \
-                    f'type of model in file: {v}'
+                msg = 'Current model type: {}, '.format(type(self).__name__) + \
```
Comment (trivialfis) [inline python-package/xgboost/sklearn.py]: Thanks for fixing this!

File: tests/ci_build/test_python.sh [hunk 2]
```diff
@@ -39,17 +43,25 @@ case "$suite" in
...
+    install_xgboost
     pytest -v -s --fulltrace tests/python
     cd tests/distributed
     ./runtests.sh
     ;;
 
+  cpu-py35)
+    source activate py35
+    install_xgboost
+    pytest -v -s --fulltrace tests/python
```
Comment (trivialfis) [inline tests/ci_build/test_python.sh]: Wouldn't this invoke the dask test?

File: python-package/xgboost/__init__.py [hunk 3]
```diff
@@ -19,6 +21,12 @@
 except ImportError:
     pass
 
+if sys.version_info[:2] == (3, 5):
+    warnings.warn(
```
Comment (trivialfis) [inline python-package/xgboost/__init__.py]: We can make a schedule.  As 35 is EOL at Sep, we can target 1.2 release assuming 2 months release.

File: tests/ci_build/test_python.sh (hunk 2 above)
Comment (hcho3) [reply tests/ci_build/test_python.sh]: Dask tests are skipped if Dask is not installed on the system.

File: python-package/xgboost/__init__.py (hunk 3 above)
Comment (hcho3) [reply python-package/xgboost/__init__.py]: Let us open a new thread to discuss when we should drop Python 3.5.

**Comment by mli** [comment]
# [Codecov](https://codecov.io/gh/dmlc/xgboost/pull/5336?src=pr&el=h1) Report
> Merging [#5336](https://codecov.io/gh/dmlc/xgboost/pull/5336?src=pr&el=desc) into [master](https://codecov.io/gh/dmlc/xgboost/commit/8aa8ef10317cfc6d625050d98a5b86d3b1b888d3&el=desc) will **decrease** coverage by `0.01%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/dmlc/xgboost/pull/5336?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master    #5336      +/-   ##
==========================================
- Coverage   83.76%   83.75%   -0.02%     
==========================================
  Files          11       11              
  Lines        2409     2413       +4     
==========================================
+ Hits         2018     2021       +3     
- Misses        391      392       +1     
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/dmlc/xgboost/pull/5336?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/dmlc/xgboost/pull/5336?src=pr&el=footer). Last update [8aa8ef1...75ca766](https://codecov.io/gh/dmlc/xgboost/pull/5336?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Pull request merged**